source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4351,6 +4351,148 @@ export class ColumnSet extends CardElementContainer {
     }
 }
 
+export class Source {
+    mimeType: string;
+    url: string;
+
+    parse(json: any): Source {
+        this.mimeType = json["mimeType"];
+        this.url = json["url"];
+        return this;
+    }
+
+    toJSON() {
+        return { mimeType: this.mimeType, url: this.url };
+    }
+}
+
+export class Media extends CardElement {
+    poster: string;
+    altText: string;
+    sources: Array<Source> = [];
+    private _mediaType: string;
+    private _mediaElement: HTMLMediaElement;
+    static readonly acceptedMediaTypes = ["audio", "video"];
+
+    private processSources(): Array<Source> {
+        let typeFilter = (t): boolean => {
+            if (Media.acceptedMediaTypes.indexOf(t)) {
```

Comment:
Does this actually work? indexOf will return 0 for the first element in the array, right? Won't that eval to false when it should eval to true?

Reply:
You are right, index 0 is falsey, so the code is wrong.
My idea was to first create some tests for the rules on audio/video and accepted types, then test it.